dali/kernels/normalize/normalize_gpu.h

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_NORMALIZE_NORMALIZE_GPU_H_
+#define DALI_KERNELS_NORMALIZE_NORMALIZE_GPU_H_
+
+/**
+ * @file
+ *
+ * This file contains the GPU kernel for directional data normalization.
+ */
+
+#include <memory>
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * @brief Normalizes data
+ *
+ * Data normalization is done using externally provided base (typically: mean or min) and scale
+ * (typically reciprocal of standard deviation or 1/(max-min)).
+ * The normalization follows the formula:
+ * ```
+ * out[data_idx] = (in[data_idx] - base[param_idx]) * scale[param_idx] * global_scale + shift
```

Comment:
Maybe you can write how `param_idx` is calculated.

Reply:
I see it is described in L54. Still small info hear would help.

Reply:
I explained it in lines 51-57.